ctf/org.eclipse.tracecompass.ctf.core/src/org/eclipse/tracecompass/internal/ctf/core/utils/LEB128.java

```diff
@@ -0,0 +1,117 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Ericsson
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License 2.0 which
+ * accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.tracecompass.internal.ctf.core.utils;
+
+import java.io.DataInput;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+/**
+ * LEB128 decoding utility functions. 
+ * Original implementation: https://git.eclipse.org/r/c/tracecompass.incubator/org.eclipse.tracecompass.incubator/+/200680
+ *
+ * @author Matthew Khouzam
+ * @author Vlad Arama
+ *
+ */
+public class LEB128 {
+    /**
+     * Reads an unsigned LEB128 encoded integer from a ByteBuffer.
+     *
+     * @param in
+     *            ByteBuffer containing the LEB128 encoded integer.
+     * @return The decoded integer as a long.
+     */
+    public static long readUnsignedLeb(ByteBuffer in) {
+        long result = 0;
+        long shift = 0;
+        byte current = 0;
+        do {
+            current = in.get();
+            result |= ((long) (current & 0x7f)) << shift;
+            shift += 7;
+        } while ((current & 0x80) != 0);
+        return result;
+    }
+
+    /**
+     * Reads an unsigned LEB128 encoded integer from a DataInput.
+     *
+     * @param in
+     *            DataInput containing the LEB128 encoded integer.
+     * @return The decoded integer as a long.
+     * @throws IOException
+     *             if the file ends
+     */
+    public static long readUnsignedLeb(DataInput in) throws IOException {
+        long result = 0;
+        long shift = 0;
+        byte current = 0;
+        do {
+            current = in.readByte();
+            result |= ((long) (current & 0x7f)) << shift;
+            shift += 7;
+        } while ((current & 0x80) != 0);
+
+        return result;
+    }
+
+    /**
+     * Reads a signed LEB128 encoded integer from a ByteBuffer.
+     *
+     * @param in
+     *            ByteBuffer containing the LEB128 encoded integer.
+     * @return The decoded signed integer as a long.
+     */
+    public static long readSignedLeb(ByteBuffer in) {
+        long result = 0;
+        int shift = 0;
+        byte current;
+        do {
+            current = in.get();
+            result |= ((long) (current & 0x7f)) << shift;
+            shift += 7;
+        } while ((current & 0x80) != 0);
+
+        if ((current & 0x40) != 0) {
```

Comment:
Good catch, we might since `long` are 64 bits in Java. 
Should I add a check to make sure the shift is smaller than 64, so that the bits don't overflow and wrap around ?
@MatthewKhouzam thoughts ?

Reply:
I agree if we pass a size argument. I looked at a few implementations in non-java libraries, They have the checks if and only if we bust the size defined. I would be OK with one of the folloing: 
* just hard-coding to 64 as a design concession
* Keeping as-is, as most implementations seem to do that
* passing a size parameter